Assets/Project/Modules/Enemies/Slime/Scripts/SlimeRefactor/EnemyHealth.cs

```diff
@@ -10,6 +10,7 @@ public class EnemyHealth : MonoBehaviour, IDamageHitTarget
     {
         private HealthSystem _healthSystem;
         [SerializeField, Range(0, 100)] private int _maxHealth = 50;
+        [SerializeField] private EnemyVisuals _enemyVisuals;
```

Comment:
If we are using a mediator, we shouldn't give access to EnemyVisuals to the health script, instead the mediator should be the one who has this reference and calls OnDeathEffects() inside its OnDeath() function, for example.

The reason why you put it here is so that we don't have to repeat the lines of code for every enemy, but this goes against single responsability principle.
Instead what we could do is refactor for the IEnemyMediator interface into an abstract class AEnemyMediator, which would have the reference to this EnemyVisuals and implement the OnDeath(), etc. methods. Then subclasses of AEnemyMediator can override OnDeath() and call its base.OnDeath()

Reply:
Probably need to talk with Roger about this change